modulus/distributed/manager.py

```diff
@@ -341,7 +341,6 @@ def setup(
                 manager._local_rank = local_rank
 
             # Setup distributed process group
-            # time.sleep(1)
             dist.init_process_group(
```

Comment:
together with line 333, does that mean that we always create a process group even if world_size = 1? If yes, how is that handled internally in torch, do all the collective ops become no-ops in that case?

Reply:
Looks like pytorch doesn't convert it into a no-op, but NCCL does:
```
  In [23]: with torch.profiler.profile() as p:                                                                                                                                                                                                                                                                                  
      ...:     tensor = torch.randn(20, 10, device=manager.device)                                                                                                                                                                                                                                                              
      ...:     dist.all_reduce(tensor)                                                                                                                                                                                                                                                                                          
      ...:                                                                                                                                                                                                                                                                                                                      
                                                                                                                                                                                                                                                                                                     
  In [24]: print(p.key_averages().table(                                                                                                                                                                                                                                                                                        
      ...:     sort_by="self_cuda_time_total", row_limit=-1))                                                                                                                                                                                                                                                                   
  -------------------------------------------------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------                                                                                                                           
                                                     Name    Self CPU %      Self CPU   CPU total %     CPU total  CPU time avg     Self CUDA   Self CUDA %    CUDA total  CUDA time avg    # of Calls                                                                                                                          
  -------------------------------------------------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------                                                                                                                           
                                            aten::normal_         0.51%      54.000us        55.02%       5.859ms       5.859ms       2.000us       100.00%       2.000us       2.000us             1                                                                                                                           
  void at::native::(anonymous namespace)::distribution...         0.00%       0.000us         0.00%       0.000us       0.000us       2.000us       100.00%       2.000us       2.000us             1                                                                                                                           
                                              aten::randn         0.22%      23.000us        97.54%      10.387ms      10.387ms       0.000us         0.00%       2.000us       2.000us             1                                                                                                                           
                                              aten::empty        42.30%       4.505ms        42.30%       4.505ms       4.505ms       0.000us         0.00%       0.000us       0.000us             1                                                                                                                           
                                    cudaStreamIsCapturing         0.10%      11.000us         0.10%      11.000us       5.500us       0.000us         0.00%       0.000us       0.000us             2                                                                                                                           
                                         cudaLaunchKernel        54.43%       5.796ms        54.43%       5.796ms       5.796ms       0.000us         0.00%       0.000us       0.000us             1                                                                                                                           
                                         c10d::allreduce_         0.54%      57.000us         1.99%     212.000us     212.000us       0.000us         0.00%       0.000us       0.000us             1                                                                                                                           
                                       record_param_comms         1.42%     151.000us         1.52%     162.000us      81.000us       0.000us         0.00%       0.000us       0.000us             2                                                                                                                           
                                      cudaStreamWaitEvent         0.08%       9.000us         0.08%       9.000us       3.000us       0.000us         0.00%       0.000us       0.000us             3                                                                                                                           
                                          nccl:all_reduce         0.00%       0.000us             0     104.000us     104.000us       0.000us         0.00%       0.000us       0.000us             1                                                                                                                           
                                    cudaDeviceSynchronize         0.40%      43.000us         0.40%      43.000us      43.000us       0.000us         0.00%       0.000us       0.000us             1                                                                                                                           
  -------------------------------------------------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------                                                                                                                           
  Self CPU time total: 10.649ms                                                                                                                                                                                                                                                                                                 
  Self CUDA time total: 2.000us  
```